src/com/amazon/corretto/crypto/provider/EvpSignature.java

```diff
@@ -324,17 +328,28 @@ private EvpSignature(AmazonCorrettoCryptoProvider provider, final EvpKeyType key
         Loader.checkNativeLibraryAvailability();
         digestName_ = digestName;
 
-        signingBuffer = new InputBuffer<byte[], EvpContext, RuntimeException>(1024)
+        signingBuffer = getSigningBuffer();
+        verifyingBuffer = getVerifyingBuffer();
+    }
+
+    private InputBuffer<byte[], EvpContext, RuntimeException> getSigningBuffer() {
+        final String pssMgfMd = pssParams_ != null
+            ? Utils.jceDigestNameToAwsLcName(
+                ((MGF1ParameterSpec) pssParams_.getMGFParameters()).getDigestAlgorithm()
+            )
+            : null;
```

Comment:
I think the following might be more readable.
```suggestion
        final String pssMgfMd;
        if (pssParams != null) {
            pssMgfMd = Utils.jceDigestNameToAwsLcName(((MGF1ParameterSpec) pssParams_.getMGFParameters()).getDigestAlgorithm());
        } else {
            pssMgfMd = null;
        }
```